test/unit/slurm/test_slurm_config_generator/test_generating_slurm_config_flexible_instance_types/expected_outputs/pcluster/slurm_parallelcluster_queue6_partition.conf

```diff
@@ -0,0 +1,7 @@
+# This file is automatically generated by pcluster
+
+NodeName=queue6-st-cr1-[1-2] CPUs=16 RealMemory=237363 State=CLOUD Feature=static,cr1,gpu Gres=gpu:a10g:4
```

Comment:
The smallest GPU count was selected but the GPU Name used for the Gres property is different.

p3.8xlarge (cheaper):
    GPU Name: V100
    GPU Count: 4

g5.48xlarge:
    GPU Name: A10G
    GPU Count: 8

The min GPU count is 4 and the GPU name should be `V100`.
This seems to be a result of using the first instance type in the list when determining the GPU Type.

One way could be to get the instance type with the min GPU count first then extract the type. 

```
def _get_min_gpu_instance_type(instance_types) -> int:
    return min(instance_types, _gpu_count)

def _gpu_count(instance_type):
    ...


... 
instance_types = [...]
min_gpu_instance_type = _get_min_gpu_instance_type(instance_types)
min_gpu_count = _gpu_count(instance_type):
gpu_type = _gpu_type(min_gpu_instance_type)
```

Reply:
done